SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -822,6 +836,20 @@ void USpatialReceiver::ApplyComponentData(Worker_EntityId EntityId, Worker_Compo
 
 void USpatialReceiver::OnComponentUpdate(Worker_ComponentUpdateOp& Op)
 {
+	if (Op.update.component_id == SpatialConstants::SERVER_RPC_ENDPOINT_COMPONENT_ID ||
+		Op.update.component_id == SpatialConstants::CLIENT_RPC_ENDPOINT_COMPONENT_ID)
+	{
+		Schema_Object* FieldsObject = Schema_GetComponentUpdateFields(Op.update.schema_type);
+		if (Schema_GetBoolCount(FieldsObject, SpatialConstants::UNREAL_RPC_ENDPOINT_READY_ID) > 0)
+		{
+			bool ready = GetBoolFromSchema(FieldsObject, SpatialConstants::UNREAL_RPC_ENDPOINT_READY_ID);
+			if (ready)
+			{
+				NetDriver->RegisterListeningEntity(Op.entity_id);
```

Comment:
It won't be enough to listen for this in component update. It might have already been set when the component was added.

Reply:
But it's set to `false` by default and then set to `true` through the componentUpdate. Do you mean Spatial might batch them and unreal workers will receive a new component with `true`?

Reply:
You don't know when each worker will check out the entity. It could be that the server gets authority and has sent the update before the client gets it checked out. It's less likely that receiving it as an update but it is possible.

Reply:
In that case, I need to add this logic to `void USpatialReceiver::OnAddComponent(Worker_AddComponentOp& Op)` as well.

Reply:
Unfortunately yes.